File: 0345-Reverse-Vowels-of-a-String/0345.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public String reverseVowels(String s) {
        List<Character> vowels = new ArrayList<>(List.of('a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'));
        char[] sArray = s.toCharArray();
        int left = 0, right = sArray.length - 1;

        while (left < right) {
            if (vowels.contains(sArray[left]) && vowels.contains(sArray[right])) {
                char temp = sArray[right];
                sArray[right] = sArray[left];
                sArray[left] = temp;
                right++;
                left--;
                continue;
            }

            if (!vowels.contains(sArray[left]))
                left++;
            if (!vowels.contains(sArray[right]))
                right++;
        }

        return String.valueOf(sArray);
    }
}

File: 0345-Reverse-Vowels-of-a-String/0345.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']
        s = list(s)
        L = 0 
        R = len(s) - 1

        while L < R:
            if s[L] in vowels and s[R] in vowels:
                temp = s[R]
                s[R] = s[L]
                s[L] = temp
                L += 1
                R -= 1
                continue
            
            if s[L] not in vowels:
                L += 1
            
            if s[R] not in vowels:
                R -= 1
        
        return "".join(s)